service/tbc/peer_manager.go

```diff
@@ -5,129 +5,349 @@
 package tbc
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand/v2"
 	"net"
 	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
 )
 
 const (
-	maxPeersGood = 1 << 13
-	maxPeersBad  = 1 << 13
+	maxPeersGood = 1024
+	maxPeersBad  = 1024
+)
+
+var (
+	testnet3Seeds = []string{
+		"testnet-seed.bitcoin.jonasschnelli.ch:18333",
+		"seed.tbtc.petertodd.org:18333",
+		"seed.testnet.bitcoin.sprovoost.nl:18333",
+		"testnet-seed.bluematt.me:18333",
+	}
+	mainnetSeeds = []string{
+		"seed.bitcoin.sipa.be:8333",
+		"dnsseed.bluematt.me:8333",
+		"dnsseed.bitcoin.dashjr.org:8333",
+		"seed.bitcoinstats.com:8333",
+		"seed.bitnodes.io:8333",
+		"seed.bitcoin.jonasschnelli.ch:8333",
+	}
 )
 
 // PeerManager keeps track of the available peers and their quality.
 type PeerManager struct {
-	peersMtx    sync.RWMutex
-	peersGood   map[string]struct{}
-	peersBad    map[string]struct{}
-	goodSeenMax int // keep track of max good peers seen to prevent early purge
+	mtx sync.RWMutex
+
+	net wire.BitcoinNet // bitcoin network to connect to
+
+	want int // number of peers we want to be connected to
+
+	dnsSeeds []string // hard coded dns seeds
+	seeds    []string // seeds obtained from DNS
+
+	peers map[string]*peer // connected peers
+	good  map[string]struct{}
+	bad   map[string]struct{}
 }
 
-// newPeerManager returns a new peer manager.
-func newPeerManager() *PeerManager {
+// NewPeerManager returns a new peer manager.
+func NewPeerManager(net wire.BitcoinNet, want int) (*PeerManager, error) {
+	if want == 0 {
```

Comment:
should we check that this is `> 0` ? or make it a `uint`?

Reply:
Fixed